crates/aranya-client/tests/tests.rs

```diff
@@ -455,3 +458,170 @@ async fn test_multi_team_sync() -> Result<()> {
 
     Ok(())
 }
+
+/// Tests hello subscription functionality by demonstrating that devices can subscribe
+/// to hello notifications from peers and automatically sync when receiving notifications.
+#[test(tokio::test(flavor = "multi_thread"))]
+async fn test_hello_subscription() -> Result<()> {
+    // Set up our team context so we can run the test.
+    let mut devices = DevicesCtx::new("test_hello_subscription").await?;
+
+    // Create the initial team, and get our TeamId.
+    let team_id = devices.create_and_add_team().await?;
+
+    // Tell all peers to sync with one another, and assign their roles.
+    devices.add_all_device_roles(team_id).await?;
+
+    // Grab addresses for testing
+    let admin_addr = devices.admin.aranya_local_addr().await?;
+
+    // Grab the shorthand for the teams we need to operate on.
+    let membera_team = devices.membera.client.team(team_id);
+    let admin_team = devices.admin.client.team(team_id);
+
+    let sync_config = SyncPeerConfig::builder()
+        .interval(Duration::from_secs(24 * 60 * 60)) // Long interval to ensure sync on hello is triggered)
+        .sync_now(false)
+        .sync_on_hello(true)
+        .build()?;
+
+    membera_team.add_sync_peer(admin_addr, sync_config).await?;
+    info!("membera added admin as sync peer with sync_on_hello=true");
+
+    // MemberA subscribes to hello notifications from Admin
+    membera_team
+        .sync_hello_subscribe(
+            admin_addr,
+            Duration::from_millis(100),
+            Duration::from_millis(1000),
+        ) // Short delay for faster testing
+        .await?;
+    info!("membera subscribed to hello notifications from admin");
+
+    // Wait a moment to ensure the subscription is active
+    common::sleep(Duration::from_millis(200)).await;
```

Comment:
Let's assign variable names rather than hard-coding these durations:
```suggestion
    let delay = Duration::from_millis(100);
    let duration = Duration::from_millis(1000);
    membera_team
        .sync_hello_subscribe(
            admin_addr,
            delay,
            duration,
        ) // Short delay for faster testing
        .await?;
    info!("membera subscribed to hello notifications from admin");

    // Wait a moment to ensure the subscription is active
    common::sleep(2 * delay).await;
```
Makes it easier to sleep based on the `delay` when waiting for a sync based on receiving a hello message.